alacritty_terminal/src/term/mod.rs

```diff
@@ -954,7 +959,16 @@ impl<T> Term<T> {
             self.mode.remove(TermMode::ALTERNATE_SCROLL);
         }
         self.default_cursor_style = config.cursor.style;
+
+        // Update title
         self.dynamic_title = config.dynamic_title();
+        if !self.dynamic_title
+            || (self.title.is_none() && self.default_title != config.window.title)
```

Comment:
The right expression `self.title.is_none()` is always true for `dynamic_title: false`, during toggle of dynamic title, title should be reset to default. In this way you don't need `Option<String>` at all.

Reply:
We don't do `!dynamic_title &&`, but `!dynamic_title ||`. That's a big difference.

Your argument for not needing `Option<String>` at all is incorrect and its introduction was completely unrelated to `dynamic_title`.

Reply:
Let's keep this discussion together under https://github.com/alacritty/alacritty/pull/3427#discussion_r391945401. But I agree.

Reply:
didn't noticed `||`.

I don't understand what are you trying to achieve.

I'm expecting reload to work in the following way.

if we do a toggle from `dynamic_title: false` to `dynamic_title: true`, set `title` to `default title`, send new `default title`.
if we do a toggle from `dynamic_title: true` to `dynamic_title: false`, set `title` to `default title`, clear title stack, send new title.
if you reload title, just set `default title` and `title` and send `default title`.

Reply:
@kchibisov who does setting `dynamic_title` from `false` to `true` need to update the title? Shouldn't it already be the default, since there's no way to change it? Otherwise I think we're on the same page.

Basically, the behavior of this PR today makes sense to me. My contention is with the implementation.

Reply:
>  if you reload title, just set default title and title and send default title.

That's exactly what I'm doing here though. If the current title is not changed (see `None` check) and the title has been changed (see `default != config.title`), then we set the title.

That's exactly what's happening here.